modules/protect.php

```diff
@@ -337,24 +339,53 @@ function check_login_ability( $preauth = false ) {
 		}
 
 		if ( isset( $transient_value ) && 'blocked' == $transient_value['status'] ) {
-			// There is a current block -- prevent login
+			$this->block_with_math();
+		}
+		
+		if ( isset( $transient_value ) && 'blocked-hard' == $transient_value['status'] ) {
 			$this->kill_login();
 		}
 
 		// If we've reached this point, this means that the IP isn't cached.
 		// Now we check with the Protect API to see if we should allow login
 		$response = $this->protect_call( $action = 'check_ip' );
-
+		
 		if ( isset( $response['math'] ) && ! function_exists( 'brute_math_authenticate' ) ) {
 			include_once dirname( __FILE__ ) . '/protect/math-fallback.php';
+			new Jetpack_Protect_Math_Authenticate;
+			return false;
 		}
 
 		if ( 'blocked' == $response['status'] ) {
+			$this->block_with_math();
+		}
+		
+		if ( 'blocked-hard' == $response['status'] ) {
 			$this->kill_login();
 		}
 
 		return true;
 	}
+	
+	function block_with_math() {
+		/**
+		 * By default, Jetpack Protect will allow a user who has been blocked for too
+		 * many failed logins to start answering math questions to continue logging in
+		 *
+		 * For added security, you can disable this 
+		 *
+		 * @since 3.6
+		 * 
+		 * @param bool Whether to allow math for blocked users or not.
+		 */
+		$allow_math_fallback_on_fail = apply_filters( 'jpp_use_captcha_when_blocked', true );
```

Comment:
Protect is one of the only modules to use prefixes other than `jetpack_`, but since `jpp_` is already being used in other hooks throughout the module, might as well keep it consistent.